test/src/unit-filter-pipeline.cc

```diff
@@ -4046,6 +4048,189 @@ TEMPLATE_TEST_CASE(
     int16_t,
     int32_t,
     int64_t) {
-  testing_float_scaling_filter<float, TestType>();
-  testing_float_scaling_filter<double, TestType>();
-}
\ No newline at end of file
+  auto negative = GENERATE(true, false);
+  testing_float_scaling_filter<float, TestType>(negative);
+  testing_float_scaling_filter<double, TestType>(negative);
+}
+
+template <typename FloatingType, typename IntType>
+void testing_float_scaling_filter_zeros(FloatingType zero_val) {
+  tiledb::sm::Config config;
+
+  // Set up test data
+  const uint64_t nelts = 100;
+  const uint64_t tile_size = nelts * sizeof(FloatingType);
+  const uint64_t cell_size = sizeof(FloatingType);
+  const uint32_t dim_num = 0;
+
+  Tile tile;
+  Datatype t = Datatype::FLOAT32;
+  switch (sizeof(FloatingType)) {
+    case 4: {
+      t = Datatype::FLOAT32;
+    } break;
+    case 8: {
+      t = Datatype::FLOAT64;
+    } break;
+    default: {
+      INFO(
+          "testing_float_scaling_filter: passed floating type with size of not "
+          "4 bytes or 8 bytes.");
+      CHECK(false);
+    }
+  }
+
+  tile.init_unfiltered(
+      constants::format_version, t, tile_size, cell_size, dim_num);
+
+  double scale = 2.53;
+  double foffset = 0.31589;
```

Comment:
Why test with these specific scale and add offsets?

Reply:
see previous comment